### PR TITLE
More retries in the build pipeline

### PR DESCRIPTION
**Pull request by pierotibou**
Copied #2149 and applied the same mechanism on installer smoke tests

@DataDog/apm-dotnet

### CODE REVIEW

**Comment by andrewlock** [comment]
## Code Coverage Report :bar_chart:

:heavy_check_mark: Merging #2226 into [master](https://github.com/DataDog/dd-trace-dotnet/tree/207d91a38b7e0c9f64a96fe8a5a25d57eaa747da) will **not change** line coverage
:heavy_check_mark: Merging #2226 into [master](https://github.com/DataDog/dd-trace-dotnet/tree/207d91a38b7e0c9f64a96fe8a5a25d57eaa747da) will **not change** branch coverage
:heavy_check_mark: Merging #2226 into [master](https://github.com/DataDog/dd-trace-dotnet/tree/207d91a38b7e0c9f64a96fe8a5a25d57eaa747da) will **not change** complexity


|           | [master](https://github.com/DataDog/dd-trace-dotnet/tree/207d91a38b7e0c9f64a96fe8a5a25d57eaa747da) | #2226       | Change   | 
|:----------|:-----------:|:-----------:|:--------:|
| Lines     | `8963` / `12180` | `8957` / `12180` |          |
| Lines %   | `74%`      | `74%`      |  `0%` :heavy_check_mark:  |
| Branches  | `4093` / `5911` | `4087` / `5911` |          |
| Branches %| `69%`      | `69%`      |  `0%` :heavy_check_mark:  |
| Complexity|   `7095`      | `7095`        |  `0`  :heavy_check_mark:    |

View the full report for further details:

* [HTML report for master](https://dev.azure.com/datadoghq/dd-trace-dotnet/_build/results?buildId=84700&view=codecoverage-tab) | [Source file](https://dev.azure.com/datadoghq/a51c4863-3eb4-4c5d-878a-58b41a049e4e/_apis/build/builds/84700/artifacts?artifactName=Code%20Coverage%20Report_84700&api-version=6.0&%24format=zip)  
* [HTML report for this PR #2226](https://dev.azure.com/datadoghq/dd-trace-dotnet/_build/results?buildId=84719&view=codecoverage-tab) | [Source file](https://dev.azure.com/datadoghq/a51c4863-3eb4-4c5d-878a-58b41a049e4e/_apis/build/builds/84719/artifacts?artifactName=Code%20Coverage%20Report_84719&api-version=6.0&%24format=zip)

### Datadog.Trace Breakdown :heavy_check_mark:

|        | [master](https://github.com/DataDog/dd-trace-dotnet/tree/207d91a38b7e0c9f64a96fe8a5a25d57eaa747da) | #2226       | Change   | 
|:-------|:-----------:|:-----------:|:--------:|
| Lines %| `74%`      | `74%`       |  `0%` :heavy_check_mark:  |
| Branches %| `69%`      | `69%`       |  `0%` :heavy_check_mark:  |
| Complexity| `7095`      | `7095`       |  `0` :heavy_check_mark:  |

The following classes have significant coverage changes.

| File    | Line coverage change | Branch coverage change | Complexity change |
|:--------|:--------------------:|:----------:|:--------:|
| [Datadog.Trace.ClrProfiler.AutoInstrumentation.Couchbase.IIOServiceExecuteIntegrationBis](https://github.com/DataDog/dd-trace-dotnet/tree/bef0cf7283595f09ff79d368424944a96cdf10b0/tracer/src/Datadog.Trace/ClrProfiler/AutoInstrumentation/Couchbase/IIOServiceExecuteIntegrationBis.cs) | `-100%` :no_entry: | `0%` :heavy_check_mark:   | `0` :heavy_check_mark: |
| [Datadog.Trace.ClrProfiler.AutoInstrumentation.Testing.MsTestV2.TestMethodRunnerExecuteIntegration](https://github.com/DataDog/dd-trace-dotnet/tree/bef0cf7283595f09ff79d368424944a96cdf10b0/tracer/src/Datadog.Trace/ClrProfiler/AutoInstrumentation/Testing/MsTestV2/TestMethodRunnerExecuteIntegration.cs) | `-33%` :no_entry: | `-19%` :no_entry:   | `0` :heavy_check_mark: |
| [Datadog.Trace.Ci.Agent.CIAgentWriter](https://github.com/DataDog/dd-trace-dotnet/tree/bef0cf7283595f09ff79d368424944a96cdf10b0/tracer/src/Datadog.Trace/Ci/Agent/CIAgentWriter.cs) | `-10%` :no_entry: | `-21%` :no_entry:   | `0` :heavy_check_mark: |
| [Datadog.Trace.ClrProfiler.AutoInstrumentation.Testing.NUnit.NUnitCompositeWorkItemSkipChildrenIntegration](https://github.com/DataDog/dd-trace-dotnet/tree/bef0cf7283595f09ff79d368424944a96cdf10b0/tracer/src/Datadog.Trace/ClrProfiler/AutoInstrumentation/Testing/NUnit/NUnitCompositeWorkItemSkipChildrenIntegration.cs) | `-8%` :no_entry: | `-8%` :no_entry:   | `0` :heavy_check_mark: |
| [Datadog.Trace.PlatformHelpers.HostMetadata](https://github.com/DataDog/dd-trace-dotnet/tree/bef0cf7283595f09ff79d368424944a96cdf10b0/tracer/src/Datadog.Trace/PlatformHelpers/HostMetadata.cs) | `6%` :heavy_check_mark: | `0%` :heavy_check_mark:   | `0` :heavy_check_mark: |


View the full reports for further details:

* [HTML report for master](https://dev.azure.com/datadoghq/dd-trace-dotnet/_build/results?buildId=84700&view=codecoverage-tab) | [Source file](https://dev.azure.com/datadoghq/a51c4863-3eb4-4c5d-878a-58b41a049e4e/_apis/build/builds/84700/artifacts?artifactName=Code%20Coverage%20Report_84700&api-version=6.0&%24format=zip)  
* [HTML report for this PR #2226](https://dev.azure.com/datadoghq/dd-trace-dotnet/_build/results?buildId=84719&view=codecoverage-tab) | [Source file](https://dev.azure.com/datadoghq/a51c4863-3eb4-4c5d-878a-58b41a049e4e/_apis/build/builds/84719/artifacts?artifactName=Code%20Coverage%20Report_84719&api-version=6.0&%24format=zip)

**Review by andrewlock — COMMENTED** [review]
Unfortunately, the command you copied has a bug 😅 any chance you could fix that one in the same PR?

File: .azure-pipelines/ultimate-pipeline.yml [hunk 1]
```diff
@@ -1692,14 +1696,18 @@ stages:
...
+            # retry up to 5 times to build the docker images
+            for i in 1 2 3 4 5; do 
+              docker build \
+                --build-arg DOTNETSDK_VERSION=$(dotnetCoreSdkLatestVersion) \
+                --build-arg RUNTIME_IMAGE=$(runtimeImage) \
+                --build-arg PUBLISH_FRAMEWORK=$(publishFramework) \
+                --build-arg INSTALL_CMD="$(installCmd)" \
+                --tag dd-trace-dotnet/$(dockerTag)-smoke-test \
+                --file "$(System.DefaultWorkingDirectory)/tracer/build/_build/docker/smoke.dockerfile" \
+                "$(smokeTestAppDir)"
```
[the remark this answers is not in the document]
Comment (pierotibou) [reply .azure-pipelines/ultimate-pipeline.yml:1708]: Sure will do :). thx

**Comment by pierotibou** [comment]
Fixed my changes only as the original command was fine, I copied/pasted a bit too quickly

File: .azure-pipelines/ultimate-pipeline.yml (hunk 1 above)
[the remark this answers is not in the document]
Comment (andrewlock) [reply .azure-pipelines/ultimate-pipeline.yml]: Oh sorry, yeah, you're right, and I misread the file view :D

**Review by andrewlock — APPROVED** [review]

**Comment by andrewlock** [comment]
## Code Coverage Report :bar_chart:

:heavy_check_mark: Merging #2226 into [master](https://github.com/DataDog/dd-trace-dotnet/tree/207d91a38b7e0c9f64a96fe8a5a25d57eaa747da) will **not change** line coverage
:heavy_check_mark: Merging #2226 into [master](https://github.com/DataDog/dd-trace-dotnet/tree/207d91a38b7e0c9f64a96fe8a5a25d57eaa747da) will **not change** branch coverage
:heavy_check_mark: Merging #2226 into [master](https://github.com/DataDog/dd-trace-dotnet/tree/207d91a38b7e0c9f64a96fe8a5a25d57eaa747da) will **not change** complexity


|           | [master](https://github.com/DataDog/dd-trace-dotnet/tree/207d91a38b7e0c9f64a96fe8a5a25d57eaa747da) | #2226       | Change   | 
|:----------|:-----------:|:-----------:|:--------:|
| Lines     | `8963` / `12180` | `8956` / `12180` |          |
| Lines %   | `74%`      | `74%`      |  `0%` :heavy_check_mark:  |
| Branches  | `4093` / `5911` | `4088` / `5911` |          |
| Branches %| `69%`      | `69%`      |  `0%` :heavy_check_mark:  |
| Complexity|   `7095`      | `7095`        |  `0`  :heavy_check_mark:    |

View the full report for further details:

* [HTML report for master](https://dev.azure.com/datadoghq/dd-trace-dotnet/_build/results?buildId=84700&view=codecoverage-tab) | [Source file](https://dev.azure.com/datadoghq/a51c4863-3eb4-4c5d-878a-58b41a049e4e/_apis/build/builds/84700/artifacts?artifactName=Code%20Coverage%20Report_84700&api-version=6.0&%24format=zip)  
* [HTML report for this PR #2226](https://dev.azure.com/datadoghq/dd-trace-dotnet/_build/results?buildId=84735&view=codecoverage-tab) | [Source file](https://dev.azure.com/datadoghq/a51c4863-3eb4-4c5d-878a-58b41a049e4e/_apis/build/builds/84735/artifacts?artifactName=Code%20Coverage%20Report_84735&api-version=6.0&%24format=zip)

### Datadog.Trace Breakdown :heavy_check_mark:

|        | [master](https://github.com/DataDog/dd-trace-dotnet/tree/207d91a38b7e0c9f64a96fe8a5a25d57eaa747da) | #2226       | Change   | 
|:-------|:-----------:|:-----------:|:--------:|
| Lines %| `74%`      | `74%`       |  `0%` :heavy_check_mark:  |
| Branches %| `69%`      | `69%`       |  `0%` :heavy_check_mark:  |
| Complexity| `7095`      | `7095`       |  `0` :heavy_check_mark:  |

The following classes have significant coverage changes.

| File    | Line coverage change | Branch coverage change | Complexity change |
|:--------|:--------------------:|:----------:|:--------:|
| [Datadog.Trace.ClrProfiler.AutoInstrumentation.Couchbase.IIOServiceExecuteIntegrationBis](https://github.com/DataDog/dd-trace-dotnet/tree/64e691477fb9411a5f7e8f335b02960987a2ca1d/tracer/src/Datadog.Trace/ClrProfiler/AutoInstrumentation/Couchbase/IIOServiceExecuteIntegrationBis.cs) | `-100%` :no_entry: | `0%` :heavy_check_mark:   | `0` :heavy_check_mark: |
| [Datadog.Trace.ClrProfiler.AutoInstrumentation.Testing.MsTestV2.TestMethodRunnerExecuteIntegration](https://github.com/DataDog/dd-trace-dotnet/tree/64e691477fb9411a5f7e8f335b02960987a2ca1d/tracer/src/Datadog.Trace/ClrProfiler/AutoInstrumentation/Testing/MsTestV2/TestMethodRunnerExecuteIntegration.cs) | `-33%` :no_entry: | `-19%` :no_entry:   | `0` :heavy_check_mark: |
| [Datadog.Trace.Ci.Agent.CIAgentWriter](https://github.com/DataDog/dd-trace-dotnet/tree/64e691477fb9411a5f7e8f335b02960987a2ca1d/tracer/src/Datadog.Trace/Ci/Agent/CIAgentWriter.cs) | `-10%` :no_entry: | `-21%` :no_entry:   | `0` :heavy_check_mark: |
| [Datadog.Trace.ClrProfiler.AutoInstrumentation.Testing.NUnit.NUnitCompositeWorkItemSkipChildrenIntegration](https://github.com/DataDog/dd-trace-dotnet/tree/64e691477fb9411a5f7e8f335b02960987a2ca1d/tracer/src/Datadog.Trace/ClrProfiler/AutoInstrumentation/Testing/NUnit/NUnitCompositeWorkItemSkipChildrenIntegration.cs) | `-8%` :no_entry: | `-8%` :no_entry:   | `0` :heavy_check_mark: |
| [Datadog.Trace.PlatformHelpers.HostMetadata](https://github.com/DataDog/dd-trace-dotnet/tree/64e691477fb9411a5f7e8f335b02960987a2ca1d/tracer/src/Datadog.Trace/PlatformHelpers/HostMetadata.cs) | `6%` :heavy_check_mark: | `0%` :heavy_check_mark:   | `0` :heavy_check_mark: |


View the full reports for further details:

* [HTML report for master](https://dev.azure.com/datadoghq/dd-trace-dotnet/_build/results?buildId=84700&view=codecoverage-tab) | [Source file](https://dev.azure.com/datadoghq/a51c4863-3eb4-4c5d-878a-58b41a049e4e/_apis/build/builds/84700/artifacts?artifactName=Code%20Coverage%20Report_84700&api-version=6.0&%24format=zip)  
* [HTML report for this PR #2226](https://dev.azure.com/datadoghq/dd-trace-dotnet/_build/results?buildId=84735&view=codecoverage-tab) | [Source file](https://dev.azure.com/datadoghq/a51c4863-3eb4-4c5d-878a-58b41a049e4e/_apis/build/builds/84735/artifacts?artifactName=Code%20Coverage%20Report_84735&api-version=6.0&%24format=zip)

**Pull request merged**